plugins/collapsible-layers.js

```diff
@@ -14,26 +14,33 @@ collapsibleLayers.initial = true; // collapsed
 function setup () {
   var CollapsibleLayers = L.LayerGroup.extend({
     initialize: function (match, options) {
-      this._ctrl = window.layerChooser;
-
-      L.Util.setOptions(this, options);
+      options = options || {};
       options.notPersistent = true;
+      L.Util.setOptions(this, options);
+
       this._name = options.name || match;
+      this._ctrl = window.layerChooser;
       this._layers = {};
       this._allLayers = [];
       if (match) { this.addLayers(match); }
+
+      this.on('add remove', function (e) { // store status
+        this.eachLayer(function (layer) {
+          this._storeOverlayState(layer._name, e.type === 'add'); //??
```

Comment:
can this be done by adding/removing the layers from the map ?

Reply:
Unfortunately not, as layers are not tracked anymore after they removed from layerChooser.

But I do not like using private methods as well, so suggestions are welcome.

Reply:
Atm I'm considering another approach: additional 'active' argument to addOverlay/BaseLayer (to enforce desirable state despite of what is recorded in last status), and use that on layers re-adding.

But it will not help to store states while layers are collapsed (e.g.: set 'Portals' to unchecked, then reload - and they are checked again).
Pretty minor case, but I'm still not completely sure.

Reply:
what about not storing anything ? 
the point of the layer is to enable/disable all collapsed layer in one click. 
It's fair not to be persistent changes

Reply:
Consider such case:
- collapse all-enabled levels into single entry
- disable that entry
- expand

What state do you expect?
Without storing state - all be (unexpectedly) enabled.

Reply:
this can be handled line 93, calling `showLayer(layer, false)` for disabled layer

Reply:
That is far from optimal: thousands of layers will be added and then immediately removed.

Reply:
Experimental: https://github.com/IITC-CE/ingress-intel-total-conversion/pull/486/commits/a32772057504212b743b6721734e773c43460e7e